share/revng/test/tests/analysis/CollectCFG/x86_64/switch-jump-table.S.filecheck

```diff
@@ -0,0 +1,15 @@
+#
+# Copyright (c) rev.ng Labs Srl. See LICENSE.md for details.
+#
+
+CHECK: define i64 @local_myfunc(i64 %rax_x86_64)
+CHECK: %1 = inttoptr i64 %rax_x86_64 to ptr
+CHECK: %2 = load i64, ptr %1, align 1
+CHECK: %3 = icmp ugt i64 %2, 5
+CHECK: switch i64 %2
+CHECK: i64 {{[0-5]+}}, label
+CHECK: i64 {{[0-5]+}}, label
+CHECK: i64 {{[0-5]+}}, label
+CHECK: i64 {{[0-5]+}}, label
+CHECK: i64 {{[0-5]+}}, label
+CHECK: i64 {{[0-5]+}}, label
```

Comment:
I think you can hardcode these, no need for a regular expression.
In fact, this is the only thing I would test: let's check that there's a `switch` statement with the exact number of case and values we expect (and not more/less).
Testing the rest is just making the test more fragile, for instance, we're going to drop `_myfunc` from the function name in favor of the `MetaAddress` of the function.

Reply:
I see OK. I will test the switch only. In general, I'm using the regexp since I cannot rely on the order of the cases.

Reply:
You can use `CHECK-DAG` then I think

Reply:
Not resolved